src/System.Runtime.Extensions/tests/System/IO/PathTests.cs

```diff
@@ -456,7 +456,16 @@ public static void GetInvalidPathChars_Core()
                 Assert.Equal(string.Empty, Path.GetExtension(bad));
                 Assert.Equal(bad, Path.GetFileName(bad));
                 Assert.Equal(bad, Path.GetFileNameWithoutExtension(bad));
-                AssertExtensions.Throws<ArgumentException>(c == 124 ? null : "path", null, () => Path.GetFullPath(bad));
+                if (c == '\0')
+                {
+                    AssertExtensions.Throws<ArgumentException>("path", null, () => Path.GetFullPath(bad));
+                }
+                else
+                {
+                    string path = Path.GetFullPath(bad);
+                    Assert.Contains(bad, path);
+                    Assert.Throws<IOException>(() => Directory.CreateDirectory(path));
```

Comment:
This line shows although GetFullPath wont throw but creating a directory with that char will throw